### PR TITLE
Log current thread id and name as span attribute

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Fixes https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/942

Changes:
* Bump opentelementry-java versions
* Fix all compilation issues
* Add a new `SpanProcessor` that appends thread attributes to each span

### CODE REVIEW

**Comment by mateuszrzeszutek** [comment]
One question to the reviewers: do you think that this new processor should be enabled/disabled with a configuration flag?

File: javaagent-tooling/src/test/groovy/io/opentelemetry/javaagent/tooling/AddThreadDetailsSpanProcessorTest.groovy
Comment (iNikem) [inline javaagent-tooling/src/test/groovy/io/opentelemetry/javaagent/tooling/AddThreadDetailsSpanProcessorTest.groovy:29]: use `expect:` and without `assert`

**Comment by iNikem** [comment]
Current snapshot of otel-java is broken. Waiting for https://github.com/open-telemetry/opentelemetry-java/pull/1585 and https://github.com/open-telemetry/opentelemetry-java/pull/1579, then I was going to merge https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/1057

File: javaagent-tooling/src/test/groovy/io/opentelemetry/javaagent/tooling/AddThreadDetailsSpanProcessorTest.groovy
Comment (mateuszrzeszutek) [reply javaagent-tooling/src/test/groovy/io/opentelemetry/javaagent/tooling/AddThreadDetailsSpanProcessorTest.groovy]: Fixed, thanks!

**Comment by mateuszrzeszutek** [comment]
> Current snapshot of otel-java is broken. Waiting for open-telemetry/opentelemetry-java#1585 and open-telemetry/opentelemetry-java#1579, then I was going to merge #1057

OK -- I'll wait for those fixes and rebase once they're merged.

**Comment by iNikem** [comment]
@mateuszrzeszutek please rebase

**Review by iNikem — APPROVED** [review]

**Comment by iNikem** [comment]
> One question to the reviewers: do you think that this new processor should be enabled/disabled with a configuration flag?

I think there is no need for such configuration. @trask do you care about that?

**Pull request merged**
